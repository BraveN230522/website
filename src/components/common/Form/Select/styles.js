import styled from '@emotion/styled';
import Icon from '@components/common/Icon';

import chevronDown from '@public/svg/chevron-down.svg';

const SelectStyles = styled.div`
  position: relative;
`;

const SelectIcon = styled(Icon)`
  position: absolute;
  left: 19px;
  top: 12px;
  cursor: pointer;
`;

const SelectLabel = styled.label`
  font-size: 18px;
  line-height: 24px;
  letter-spacing: -0.12px;
  margin-left: 9px;
`;

const SelectComponent = styled.select`
  appearance: none;
  background-image: url(${chevronDown});
  background-position: 95% 50%;
  background-repeat: no-repeat;
  background-color: transparent;
  border: 2px solid ${({ theme }) => theme.colors.white};
  border-radius: 25px;
  color: ${({ theme }) => theme.colors.white};
  display: block;
  font-size: 16px;
  cursor: pointer;
  font-weight: 600;
  outline: none;
  width: 100%;
  height: 45px;
  padding: 0rem 3rem;
`;

const ErrorContainer = styled.div`
  margin-top: 5px;
`;

const ErrorMessage = styled.span`
  color: ${({ theme }) => theme.colors.error};
  font-weight: 600;
  display: block;
  font-size: 12px;
  padding-left: 20px;
`;

export { SelectComponent, ErrorContainer, SelectStyles, SelectIcon, ErrorMessage, SelectLabel };
