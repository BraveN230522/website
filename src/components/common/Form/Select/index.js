import React, { forwardRef } from 'react';

import { SelectComponent, SelectStyles, SelectIcon, ErrorContainer, ErrorMessage, SelectLabel } from './styles';

const Select = forwardRef(
  (
    {
      data,
      icon,
      className,
      label,
      defaultValue,
      value,
      placeholder,
      name,
      onChange,
      disableUnselect,
      onFocus,
      error,
      disabled,
      dataIndex,
    },
    ref
  ) => (
    <SelectStyles>
      {Boolean(label) && <SelectLabel htmlFor={name}>{label}</SelectLabel>}
      {!!icon && <SelectIcon name={icon} />}
      <SelectComponent
        className={className}
        defaultValue={defaultValue}
        value={value}
        name={name}
        onChange={onChange}
        onFocus={onFocus}
        ref={ref}
        disabled={disabled}
        data-index={dataIndex}
      >
        <option value="" disabled={disableUnselect}>
          {placeholder}
        </option>

        {!!data.length &&
          data.map((current, index) => (
            <option key={index} value={current.value}>
              {current.label}
            </option>
          ))}
      </SelectComponent>
      {Boolean(error) && (
        <ErrorContainer>
          <ErrorMessage>{error}</ErrorMessage>
        </ErrorContainer>
      )}
    </SelectStyles>
  )
);

export default Select;
