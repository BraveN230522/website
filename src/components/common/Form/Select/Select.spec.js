import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import Select from './index';

export const mockSelectOption = ['', 'New York', 'Kansas', 'Chicago', 'Los Angeles'];

describe('<Select />', () => {
  it('Renders the select unchanged', () => {
    const { container } = render(<Select data={mockSelectOption} />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });
});
