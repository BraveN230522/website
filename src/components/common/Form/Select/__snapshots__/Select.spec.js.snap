// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> Renders the select unchanged 1`] = `
.emotion-2 {
  position: relative;
}

.emotion-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-image: url(mocked-file);
  background-position: 95% 50%;
  background-repeat: no-repeat;
  background-color: transparent;
  border: 2px solid #ffffff;
  border-radius: 25px;
  color: #ffffff;
  display: block;
  font-size: 16px;
  cursor: pointer;
  font-weight: 600;
  outline: none;
  width: 100%;
  height: 45px;
  padding: 0rem 3rem;
}

<div>
  <div
    class="emotion-2 emotion-3"
  >
    <select
      class="emotion-0 emotion-1"
    >
      <option
        value=""
      />
      <option />
      <option />
      <option />
      <option />
      <option />
    </select>
  </div>
</div>
`;
